Source/Syrup/Player/SyrupPlayerCharacter.cpp

```diff
@@ -71,6 +83,21 @@ void ASyrupPlayerCharacter::MoveRight(float AxisValue)
 	GetCharacterMovement()->AddInputVector(FVector(0.f, AxisValue, 0.f));
 }
 
+
+void ASyrupPlayerCharacter::Plant()
+{
+	APlant::SowPlant(this, Energy, PlantTypes[PlantSelected], (ASyrupPlayerCharacter::GetTransform()));
+
+	UE_LOG(LogTemp, Warning, TEXT("PLANT"));
+}
+
+void ASyrupPlayerCharacter::EndTurn()
+{
+	Energy = 10;
```

Comment:
Bad hard coding, should be reset to the class default value. You can get it by doing 
`Cast<ASyrupPlayerCharacter>(StaticClass()->GetDefaultObject())->Energy `